src/app/donation-start/donation-start.component.ts

```diff
@@ -312,21 +312,12 @@ export class DonationStartComponent implements AfterContentChecked, OnDestroy, O
       }
     }, 200);
 
-    // This ensures `stepper.reset()` evalutes this to false and our usage
-    // with this const helps to prevent `createDonation()` from being incorrectly triggered.
-    const activelySelectedNext = (
-      event.previouslySelectedStep.label === 'Your donation' &&
-      event.previouslySelectedStep.interacted === true
-    );
-
-    const invalidDonation = (this.donation === undefined || this.donation.status === 'Cancelled');
-
-    const invalidPreviousDonation = (this.previousDonation === undefined || this.previousDonation.status === 'Cancelled');
-
-    // Create a donation if user actively clicks 'next' from first step and
-    // the current and previous donations are invalid.
-    if (activelySelectedNext && invalidDonation && invalidPreviousDonation) {
-      this.createDonation();
+    // Create a donation if coming from first step and not offering to resume
+    // an existing donation.
+    if (event.previouslySelectedStep.label === 'Your donation') {
```

Comment:
@NoelLH removing the `event.previouslySelectedStep.interacted === true` check here has caused donations that are cancelled through this modal:

![image](https://user-images.githubusercontent.com/47176130/99242035-90d13580-27f6-11eb-859b-920c0c7655ca.png)

to show this error:

![image](https://user-images.githubusercontent.com/47176130/99242075-a21a4200-27f6-11eb-8027-059b0ec1ed6d.png)

because in `donation-start.component.getDialogResponseFn()` we reset the stepper and its causing it to create an invalid donation

Reply:
OK – it sounds like this is a reason to validate the state of the local vars, not the specific UI interactions which is far too brittle @tbg-jasonfung 

I'm not clear on the order of steps for this scenario, could you list them please?

Reply:
Don't worry, got it